Profiler/TriggerProfiler.md

```diff
@@ -0,0 +1,61 @@
+
+![](https://user-images.githubusercontent.com/15442480/66504286-af0a2500-ea7d-11e9-8f15-21af1eba1fa6.png)
+
+This opens up a new trigger configuration view:
+
+![](https://user-images.githubusercontent.com/15442480/66504172-7407f180-ea7d-11e9-8ef2-b7b46b181e26.png)
+
+There are 4 different options that can be configured.
+
+1. CPU Trigger allows you to turn the enable or disable the trigger. The default value is on. 
+2. CPU Threshold (%) allows you to specify the CPU threshold that has to be breached to start the profiling session. The default value is 80%.
+3. Profiling Duration allows you to specify the length of the profiling session once the threshold has been breached. The default value is 120 seconds (2 minutes).
+4. Cooldown allows you specify the amount of time that has to pass before another profiler session starts. The default value is 4 hrs. 
+
+Since we are interested in getting traces when the CPU breaches 85% we simply change the CPU Threshold value to 85 and leave the rest of the options as defaults. Next we click Apply. 
+
+At this point we have configured the Profiler to start collecting traces when the CPU breaches 85%. 
+
+
+### Viewing Traces Generated From Triggers
+At this point, we wait for the condition (85%+ CPU) to occur again and then turn our attention to the traces generated. The easiest way to view the generated traces is via the Configure Profiler button in the performance blade. On the configuration blade you will notice a list of available profiling sessions. To make it easier to understand whether a trace has been generated as a result of default sampling or a trigger a new column has been added called 'Triggered By' as shown below. 
+
+![](https://user-images.githubusercontent.com/15442480/66504241-969a0a80-ea7d-11e9-875b-65f144e60439.png)
+
+It also very conveniently tells you what the CPU and memory consumption was at the time of the profiler session. As you can see, a profiler session was started as a result of the previous CPU trigger that we configured. The CPU was at 88% and memory at 71%. We can now click on that row to open up the trace view blade and investigate further:
+
+![](https://user-images.githubusercontent.com/15442480/66504269-a3b6f980-ea7d-11e9-8563-9f880412ee98.png)
+
+The above blade is split into two primary sections:
+
+1. Examples - shows a list of traces available in the session sorted by slowest wall clock time. You can pick individual traces to view. 
+2. Trace view - shows the actual trace for the selected entry.
+
+The trace offers a view of what was actually happening in code for that particular request. In the above case we can see that a request (/perfCPU/HighCPU/10000) was invoked and took 10 seconds to complete. If we look at the actual call tree, we can see that most of that time was taken in my controller doing a Sort of some kind. I can now switch over to my favorite code editor and look at that particular piece of code to see why the slowness occurred while sorting. 
+
```

Comment:
Suggestion, what do you think about taking the chance to ask user for feedback here at the end of the post?